tests/console/redis.pm

```diff
@@ -0,0 +1,116 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: redis tests
+# - install redis and start redis-server
+# - connect to redis client and perform few CRUD ops
+# - load a test db from the data dir
+# - start another redis instance with a different port
+# - make new instance a replica of the earlier instance
+#
+# Maintainer: QE-Core <qe-core@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils 'zypper_call';
+
+sub wait_serial_or_die {
+    my ($feedback, %args) = @_;
+    $args{timeout} //= 10;
+
+    my $e = wait_serial($feedback, %args);
+    if (!defined $e) {
+        die('Unexpected serial output');
+    }
+}
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    # install redis package
+    zypper_call 'in redis';
+
+    # start redis server
+    background_script_run('redis-server');
+    wait_still_screen(5);
+
+    # get a redis client prompt
+    enter_cmd('redis-cli');
```

Comment:
Instead of "typing" all commands into the same command interface, you could also use `redis-cli COMMAND` and simpler commands like `assert_script_run` or `validate_script_output`. Something like

```suggestion
    # Test if the CLI is showing up
    enter_cmd('redis-cli');
    wait_serial_or_die('127.0.0.1:6379');
    enter_cmd('quit');
    
    # Test various redis commands
    assert_script_run("redis-cli ping");
    # ...
    validate_script_output("redis-cli pfselftest", sub { m/OK/ });
    # ...
```

Reply:
Thank you @grisu48 for reviewing and providing a nice suggestion. I have pushed the updated code, please take a look.

Reply:
Looks good!